xbmc/network/WakeOnAccess.cpp

```diff
@@ -66,6 +71,118 @@ static unsigned long HostToIP(const std::string& host)
   return inet_addr(ip.c_str());
 }
 
+#define LOCALIZED(id) g_localizeStrings.Get(id)
+
+static void ShowDiscoveryMessage(const char* function, const char* server_name, bool new_entry)
+{
+  std::string message, heading = LOCALIZED(13033);
+
+  if (new_entry)
+  {
+    CLog::Log(LOGINFO, "%s - Create new entry for host '%s'", function, server_name);
+    message = StringUtils::Format(LOCALIZED(13035).c_str(), server_name);
+  }
+  else
+  {
+    CLog::Log(LOGINFO, "%s - Update existing entry for host '%s'", function, server_name);
+    message = StringUtils::Format(LOCALIZED(13034).c_str(), server_name);
+  }
+  CGUIDialogKaiToast::QueueNotification(CGUIDialogKaiToast::Info, heading, message, 4000, true, 3000);
+}
+
+struct UPnPServer
+{
+  UPnPServer()
+  {
+    m_nextWake = CDateTime::GetCurrentDateTime();
+  }
+  bool operator == (const UPnPServer& server) const { return server.m_uuid == m_uuid; }
+  bool operator != (const UPnPServer& server) const { return server.m_uuid != m_uuid; }
```

Comment:
The best way to implement `operator!=()` is with
```
return !(*this == server);
```
because it will re-use the implementation of `operator==()` and you don't have to duplicate the actual logic.